files/check_hddtemp

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+# Author: Michel Weitbrecht (michel.weitbrecht@stuvus.uni-stuttgart.de)
+
+import os
+import subprocess
+import argparse
+import json
+from common import doCheck
+
+POSSIBLE_STATES = ('OK', 'WARNING', 'CRITICAL', 'UNKNOWN')
+
+def raiseState(newState, ret):
+    if newState not in POSSIBLE_STATES:
+        ret['state'] = 'UNKNOWN'
+    elif POSSIBLE_STATES.index(newState) > POSSIBLE_STATES.index(ret['state']):
+        ret['state'] = newState
+
+
+def check(ret):
+    ret['name'] = 'hddtemp'
+
+    parser = argparse.ArgumentParser(
+        prog='check_hddtemp',
+        description='Check the temperature of all SCSI-like devices.')
+    parser.add_argument('-tw', '--temp-warn',
+                        help="Return WARNING if the temperature in degree Celcius of any device is higher than this", type=int,
+                        default=40)
+    parser.add_argument('-tc', '--temp-crit',
+                        help="Return CRITICAL if the temperature in degree Celcius of any device is higher than this", type=int,
+                        default=50)
+    parser.add_argument('-i', '--ignore',
+                        help="Ignore this device, e.g. 'sda'. May be provided multiple times.", default=[], action='append')
+    args = vars(parser.parse_args())
+
+    lsblk_process = subprocess.Popen(['lsblk', '-SJ', '--output=NAME,MODEL'], stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    lsblk_json_output = lsblk_process.communicate()[0].decode('ISO-8859-1')
+
+    if lsblk_process.returncode is not 0:
+        raiseState('UNKNOWN', ret)
+        ret['output'] = '`lsblk -SJ --output=NAME,MODEL` failed with return code {}.'.format(lsblk_process.returncode)
+        ret['details'] = lsblk_json_output
+        return
+    elif len(lsblk_json_output) == 0:
+        ret['output'] = 'No devices to check.'
+        return
+
+    lsblk_devices = json.loads(lsblk_json_output)['blockdevices']
+    model_for_device = {}
+    for disk in lsblk_devices:
+        model_for_device[disk['name']] = disk['model']
+
+    hddtemp_command = ['hddtemp', '-n']
+    scan_devices = []
+    for device in lsblk_devices:
+        if device['name'] not in args['ignore']:
+            dev_path = '/dev/{}'.format(device['name'])
+            # Check if access to device is allowed for the executing user
+            if not os.access(dev_path, os.R_OK):
+                raiseState('UNKNOWN', ret)
+                ret['output'] = 'Not allowed to access {}'.format(dev_path)
+                return
+            scan_devices.append(dev_path)
+
+    hddtemp_process = subprocess.Popen(hddtemp_command + scan_devices, stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    hddtemp_output = hddtemp_process.communicate()[0].decode('ISO-8859-1')
+
+    min_temp = 65
```

Comment:
Can you please describe why the `min_temp` and `max_temp` are required and the usage of those?

Reply:
They're used in the final output of the check: https://github.com/stuvusIT/icinga2-scripts/pull/23/files#diff-aea7ff42554e3991b1b0e15f6777f114R88
Without these, the message would just be that the check is okay, but that information is already provided by the check status. Printing the temperature for all devices there would be cluttered and have a differing length depending on the number of devices. That's why I chose to calculate and display the coldest and hottest temperature.

Reply:
Why 65 and not something like 999?

Reply:
I think 65 is a good value for a maximum temperatur a HDD can have. If you choose to much, the pie diagrams in icingaweb2 will show mostly the same value.
I added an option to set the maximum value, though.